File: jniLibDemo/src/main/java/me/ycdev/android/lib/commonjni/demo/ResourceLimitActivity.java
package me.ycdev.android.lib.commonjni.demo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import me.ycdev.android.lib.commonjni.SysResourceLimitHelper;

public class ResourceLimitActivity extends AppCompatActivity implements View.OnClickListener {
    private TextView mOflimitStatusView;
    private Button mIncreaseOflimitBtn;
    private Button mDecreaseOflimitBtn;

    private SysResourceLimitHelper.LimitInfo mCurOflimit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_resource_limit);

        mOflimitStatusView = (TextView) findViewById(R.id.oflimit_status);
        mIncreaseOflimitBtn = (Button) findViewById(R.id.oflimit_increase);
        mIncreaseOflimitBtn.setOnClickListener(this);
        mDecreaseOflimitBtn = (Button) findViewById(R.id.oflimit_decrease);
        mDecreaseOflimitBtn.setOnClickListener(this);

        refreshOflimitStatus();
    }

    private void refreshOflimitStatus() {
        mCurOflimit = SysResourceLimitHelper.getOpenFilesLimit();
        String status = getString(R.string.oflimit_status, mCurOflimit.curLimit, mCurOflimit.maxLimit);
        mOflimitStatusView.setText(status);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_resource_limit, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onClick(View v) {
        if (v == mIncreaseOflimitBtn) {
            if (mCurOflimit.curLimit == 0) {
                mCurOflimit.curLimit = 1;
            }
            if (!SysResourceLimitHelper.setOpenFilesLimit(mCurOflimit.curLimit * 2)) {
                Toast.makeText(this, "failed to set limit", Toast.LENGTH_SHORT).show();
            }
            refreshOflimitStatus();
        } else if (v == mDecreaseOflimitBtn) {
            if (!SysResourceLimitHelper.setOpenFilesLimit(mCurOflimit.curLimit / 2)) {
                Toast.makeText(this, "failed to set limit", Toast.LENGTH_SHORT).show();
            }
            refreshOflimitStatus();
        }
    }
}

File: jniLibDemo/src/main/java/me/ycdev/android/lib/commonjni/demo/ResourceLimitActivity.kt
package me.ycdev.android.lib.commonjni.demo

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity
import me.ycdev.android.lib.commonjni.SysResourceLimitHelper

class ResourceLimitActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var ofLimitStatusView: TextView
    private lateinit var increaseOflimitBtn: Button
    private lateinit var decreaseOflimitBtn: Button

    private lateinit var curOflimit: SysResourceLimitHelper.LimitInfo

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_resource_limit)

        ofLimitStatusView = findViewById<View>(R.id.oflimit_status) as TextView
        increaseOflimitBtn = findViewById<View>(R.id.oflimit_increase) as Button
        increaseOflimitBtn.setOnClickListener(this)
        decreaseOflimitBtn = findViewById<View>(R.id.oflimit_decrease) as Button
        decreaseOflimitBtn.setOnClickListener(this)

        refreshOflimitStatus()
    }

    private fun refreshOflimitStatus() {
        curOflimit = SysResourceLimitHelper.getOpenFilesLimit()
        val status =
            getString(R.string.oflimit_status, curOflimit.curLimit, curOflimit.maxLimit)
        ofLimitStatusView.text = status
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_resource_limit, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId

        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    override fun onClick(v: View) {
        if (v === increaseOflimitBtn) {
            if (curOflimit.curLimit == 0) {
                curOflimit.curLimit = 1
            }
            if (!SysResourceLimitHelper.setOpenFilesLimit(curOflimit.curLimit * 2)) {
                Toast.makeText(this, "failed to set limit", Toast.LENGTH_SHORT).show()
            }
            refreshOflimitStatus()
        } else if (v === decreaseOflimitBtn) {
            if (!SysResourceLimitHelper.setOpenFilesLimit(curOflimit.curLimit / 2)) {
                Toast.makeText(this, "failed to set limit", Toast.LENGTH_SHORT).show()
            }
            refreshOflimitStatus()
        }
    }
}
